### PR TITLE
Added Contributed Resources Preference Pane

### LINKED ISSUE #1134: Users need a way to override contributed property sets
Contributed property sets (and packages) used to be copied into the workspace, which we considered bad practice. In #527 we changed that, such that contributed AADL files are read from the contributing plugins. However, now it is impossible to fix errors in such a file without updating OSATE. There should be some way to update just a single contributed file.

We have a specialized mechanism that works for `AADL_Project.aadl` only.

Who updates the file contents? End user or update mechanism?
What's the scope of the change? Workspace or installation?

### PR DESCRIPTION
**Pull request by AaronGreenhouse**
Fixed #1134

### CODE REVIEW

**Comment by AaronGreenhouse** [comment]
Moved the documentation to a new "Advanced OSATE Features" document and out of the "Getting Started" guide.

Removed the old "Override AADL_Project" command from the UI.

Fixed the preference pane to not show the resource or platform prefix on the URIs.

**Comment by AaronGreenhouse** [comment]
> 2. The layout and scrollability of the preference page could be improved. 

Agreed.  I had troubles getting the scroll pane to work right at the time, but I've since used it for something else and may have better luck now...

**Comment by AaronGreenhouse** [comment]
Split the preference pane "in half" using a sash, and encapsulated the the top list in a scroll pane.  Things are still a little weird because Eclipse puts the entire preference pane into another scroll pane, and the scroll panes interact strangely with each other.

**Comment by AaronGreenhouse** [comment]
Made more changes along the #1134 branch.  See the comments there.

**Comment by AaronGreenhouse** [comment]
Merged Anna's viewer changes.  I think this is ready to go!

**Comment by AaronGreenhouse** [comment]
Updated the help docs.

File: core/org.osate.help/markdown/AdvancedOSATEFeatures.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Occasionally it may be necessary to "override" a contribution, for example to customize property constant values or to correct errors.  OSATE enables this via the `OSATE > Contributed Resources` preference pane.  
+
+![Contributed Resources Preference Pane](images/AdvancedOSATEFeatures/ContributedResourcesPreferencePane.png)
+
+The top of the pane shows a tree that closely mimics the sub-tree shown below the `Plug-in Contributions` item in the `AADL Navigator`.  Selecting an item in the tree shows the Xtext URI of the contributed resource.  
+Double-clicking on an item or pressing the `Override` button brings up a selection dialog box.  This dialog is used to select an resource in the workspace that will be used to override the plug-in contributed resource.
+The overriding resource must have the same filename as the original resource; the selection dialog only shows those files in the workspace that have the same name.  The below selection dialog shows the a replacement
```
Comment (joeseibel) [inline core/org.osate.help/markdown/AdvancedOSATEFeatures.md:16]: Too many articles. Either "the" or "a" should be removed and the html should be regenerated.

```suggestion
The overriding resource must have the same filename as the original resource; the selection dialog only shows those files in the workspace that have the same name.  The below selection dialog shows the replacement 
```

File: core/org.osate.help/markdown/AdvancedOSATEFeatures.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+![Contributed Resources Preference Pane](images/AdvancedOSATEFeatures/ContributedResourcesPreferencePane.png)
+
+The top of the pane shows a tree that closely mimics the sub-tree shown below the `Plug-in Contributions` item in the `AADL Navigator`.  Selecting an item in the tree shows the Xtext URI of the contributed resource.  
+Double-clicking on an item or pressing the `Override` button brings up a selection dialog box.  This dialog is used to select an resource in the workspace that will be used to override the plug-in contributed resource.
+The overriding resource must have the same filename as the original resource; the selection dialog only shows those files in the workspace that have the same name.  The below selection dialog shows the a replacement 
+resource for `AADL_Project` being selected:
+
+![Selecting a Replacement Resource from the Workspace](images/AdvancedOSATEFeatures/Selection.png)
+
+When a contributed resource is overridden, the tree marks it as such, and the URI shows the workspace URi of the overriding resource:
```
Comment (joeseibel) [inline core/org.osate.help/markdown/AdvancedOSATEFeatures.md:21]:
```suggestion
When a contributed resource is overridden, the tree marks it as such, and the URI shows the workspace URI of the overriding resource:
```

**Review by joeseibel — CHANGES_REQUESTED** [review]
There are a couple simple English mistakes in the help text. See line specific comments.

**Review by lwrage — APPROVED** [review]

**Pull request merged**